tests/framework/e2e/etcdctl.go

```diff
@@ -176,3 +176,70 @@ func (ctl *EtcdctlV3) Compact(rev int64, o config.CompactOption) (*clientv3.Comp
 
 	return nil, SpawnWithExpect(args, fmt.Sprintf("compacted revision %v", rev))
 }
+
+func (ctl *EtcdctlV3) Status() ([]*clientv3.StatusResponse, error) {
+	args := ctl.cmdArgs()
+	args = append(args, "endpoint", "status", "-w", "json")
```

Comment:
It would be better to let users to configure the output format, and we can provide a default value.  Ideally all the [globalFlags](https://github.com/etcd-io/etcd/blob/main/etcdctl/ctlv3/ctl.go#L39) should be configurable, we can even add the field into `EtcdctlV3` (see below). Of course, it shouldn't be the scope/responsibility of this PR. 

```
type EtcdctlV3 struct {
	clusterCfg       *EtcdProcessClusterConfig
        globalCfg        command.GlobalFlags{}
}
```

Reply:
> It would be better to let users to configure the output format, and we can provide a default value. Ideally all the [globalFlags](https://github.com/etcd-io/etcd/blob/main/etcdctl/ctlv3/ctl.go#L39) should be configurable, we can even add the field into `EtcdctlV3` (see below). Of course, it shouldn't be the scope/responsibility of this PR.
> 
> ```
> type EtcdctlV3 struct {
> 	clusterCfg       *EtcdProcessClusterConfig
>         globalCfg        command.GlobalFlags{}
> }
> ```

We need to unmarshal it in line 197. and in line 182, it needs to print with json.
E2e test cases used one of format output. It should be different one by one.

Reply:
We need to unmarshal it in line 197. and in line 182, it needs to print with json.
E2e test cases used one of format output. It should be different one by one.

Reply:
@kkkkun is right, the `-w json` is a binary specific flag that is needed to parse the output to return same proto response as integration tests. This flag and other e2e specific features will be useful when we also start working on e2e specific tests (tests that cannot be replicated for integration tests, like downgrade tests as they expect to run last release binary), however for now let's focus on migrating the common subset of tests.